frontend/src/components/Users/UsersList/partials/ListOfCards/index.tsx

```diff
@@ -1,34 +1,90 @@
-import React from 'react';
+import React, { useMemo, useRef } from 'react';
 
 import { DotsLoading } from '@/components/loadings/DotsLoading';
 import Flex from '@/components/Primitives/Flex';
 import { UserWithTeams } from '@/types/user/user';
 import Text from '@/components/Primitives/Text';
 import SearchInput from '@/components/Teams/CreateTeam/ListMembers/SearchInput';
-import { useRecoilValue } from 'recoil';
-import { usersWithTeamsState } from '@/store/user/atoms/user.atom';
+import { useSetRecoilState } from 'recoil';
+// import { usersWithTeamsState } from '@/store/user/atoms/user.atom';
+import { useInfiniteQuery } from 'react-query';
+import { getAllUsersWithTeams } from '@/api/userService';
+import { ToastStateEnum } from '@/utils/enums/toast-types';
+import { toastState } from '@/store/toast/atom/toast.atom';
 import { ScrollableContent } from '../../../../Boards/MyBoards/styles';
 import CardBody from '../CardUser/CardBody';
 
-type ListOfCardsProp = {
-  isLoading: boolean;
-};
+const ListOfCards = React.memo(() => {
+  const setToastState = useSetRecoilState(toastState);
+  // const setUsersWithTeamsState = useSetRecoilState(usersWithTeamsState);
+  const scrollRef = useRef<HTMLDivElement>(null);
+
+  const fetchUsers = useInfiniteQuery(
+    'usersWithTeams',
+    ({ pageParam = 0 }) => getAllUsersWithTeams(pageParam),
+    {
+      enabled: true,
+      refetchOnWindowFocus: false,
+      getNextPageParam: (lastPage) => {
+        const { hasNextPage, page } = lastPage;
+        if (hasNextPage) return page + 1;
+        return undefined;
+      },
+      onError: () => {
+        setToastState({
+          open: true,
+          content: 'Error getting the users',
+          type: ToastStateEnum.ERROR,
+        });
+      },
+    },
+  );
+
+  const { data, isLoading } = fetchUsers;
+
+  // if (data) {
+  //   setUsersWithTeamsState(data?.pages[0].userWithTeams);
+  // }
```

Comment:
```suggestion
  // if (data) {
  //   setUsersWithTeamsState(data?.pages[0].userWithTeams);
  // }
```

If you don't need this for the next feature please remove it